### PR TITLE
new evalContext

### PR DESCRIPTION
**Pull request by FrancoisGe**

### CODE REVIEW

File: addons/web/static/src/views/relational_model/record.js [hunk 1]
```diff
@@ -228,14 +232,22 @@ export class Record extends DataPoint {
...
+        this._setDataContext();
         this._removeInvalidFields(Object.keys(changes));
+        for (const [fieldName, value] of Object.entries(this.data)) {
+            if (
+                this.fields[fieldName].type === "one2many" ||
+                this.fields[fieldName].type === "many2many"
+            ) {
+                value._updateContext(getFieldContext(this, fieldName));
+            }
+        }
```
Comment (FrancoisGe) [inline addons/web/static/src/views/relational_model/record.js]: remove

File: addons/web/static/src/views/relational_model/record.js [hunk 2]
```diff
@@ -456,9 +468,8 @@ export class Record extends DataPoint {
...
     }
 
-    _computeEvalContext() {
-        const evalContext = {
-            ...this.context,
+    _computeDataContext() {
+        const dataContext = {
             active_id: this.resId || false,
             active_ids: this.resId ? [this.resId] : [],
             active_model: this.resModel,
```
Comment (FrancoisGe) [inline addons/web/static/src/views/relational_model/record.js]: move to evalContext getter

File: addons/web/static/src/views/relational_model/record.js [hunk 3]
```diff
@@ -31,15 +31,15 @@ export class Record extends DataPoint {
         this.data = { ...this._values, ...this._changes };
         const parentRecord = this._parentRecord;
         if (parentRecord) {
-            this.evalContext = {
+            this.dataContext = {
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/record.js]: _dataContext
```suggestion
            this._dataContext = {
```

File: addons/web/static/src/views/relational_model/record.js [hunk 4]
```diff
@@ -57,6 +57,19 @@ export class Record extends DataPoint {
...
+        return Object.assign(
+            {
+                active_id: this.resId || false,
+                active_ids: this.resId ? [this.resId] : [],
+                active_model: this.resModel,
+                current_company_id: this.model.company.currentCompany.id,
+            },
+            this.context,
+            this.dataContext
+        );
```
Comment (aab-odoo) [inline addons/web/static/src/views/relational_model/record.js]: it's not equivalent to before (e.g. active_id key in context would override the active_id defined here)
```suggestion
    get evalContext() {
        return Object.assign(
            {},
            this.context,
            {
                active_id: this.resId || false,
                active_ids: this.resId ? [this.resId] : [],
                active_model: this.resModel,
                current_company_id: this.model.company.currentCompany.id,
            },
            this.dataContext
        );
```

**Review by aab-odoo — APPROVED** [review]

**Pull request merged**